app/lib/frontend/models.dart

```diff
@@ -221,6 +222,133 @@ class PackageVersion extends db.ExpandoModel {
   }
 }
 
+/// An derived entity that holds only the `pubspec.yaml` content of [PackageVersion].
+///
+/// The content of `pubspec.yaml` may be updated/cleaned in case of a breaking
+/// change was introduced since the [PackageVersion] was published.
+@db.Kind(name: 'PackageVersionPubspec', idType: db.IdType.String)
+class PackageVersionPubspec extends db.ExpandoModel {
```

Comment:
`*Info` has potentially large file content, which is not needed when we are serving the API that is used by the pub client. We can make those requests much faster if we keep only this field here.

Reply:
As I understood from @jonasfj you would anyway restrict size of the text that is actually included in the `*Info`.

Reply:
It is still 128k per file, for three files. As the API call requires us to load all of the versions for a given package, it seems to be wasteful to do so even for small files, if the number of versions are large.

Reply:
Currently the package-version contains all the fields (without length-restriction), and the performance seems fine?
I'm not vetoing anything. It just seems quite complex. Hopefully the efficiency gains are worth it.
We could also consider sticking the pubspec in the canonical info - just because it is small and useful.

Reply:
> Currently the package-version contains all the fields (without length-restriction), and the performance seems fine?

I wouldn't say it is fine... we can certainly improve the performance of that.

> We could also consider sticking the pubspec in the canonical info - just because it is small and useful.

We cannot guarantee that it is small: people can put a lot of things in the pubspec, which would be then ignored by the tools. Handling it as non-canonical would allow us to make sure that it is kept small: we could whitelist fields that we store here, we could prune it in case some of the fields have a breaking change in the parser...

Reply:
Ok - you've convinced me :)